test/setupTestCoreApi.ts

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License.
+
+import Module = require('module');
+
+export function setupTestCoreApi(): void {
```

Comment:
Not sure if I'm following this correctly, but it seems like here we're defining our own mock core API for testing purposes, which isn't necessarily the same core API provided by the worker. This means that if, for whatever reason, this API changes from the worker, we have to update it here too. I don't have ideas off the top of my head, but is there any way we can test against the actual core API directly, or link the two somehow?